azure-iot-device/tests/common/pipeline/test_pipeline_stages_mqtt.py

```diff
@@ -337,16 +304,20 @@ def test_provider_function_throws_base_exception(
 
 
 @pytest.mark.parametrize("params", connection_ops)
-@pytest.mark.describe("MQTT Provider pipeline stage _runOp function with Connect operations")
+@pytest.mark.describe(
+    "Provider - .run_op() -- called with op that connects, disconnects, or reconnects"
```

Comment:
>called with op that connects, disconnects, or reconnects [](start = 29, length = 56)

Not a problem with this test - but I do wonder if the shared handler for connect/disconnect is something we should try to get rid of at some point. I've been less than thrilled with it from the client layer. I'm guessing it's also kind of weird to work with farther down the stack, given the way this TestClass has to be awkwardly grouped around the shared functionality. Is there any good reason we're doing it this way still?

This comment inspired by this PR, but obviously not relevant to it.